go-controller/pkg/libovsdbops/db_object_ids.go

```diff
@@ -0,0 +1,295 @@
+}
+
+func (it ObjectIDsType) GetIDs() []string {
+	return it.ids
+}
+
+func (it ObjectIDsType) IsSameType(it2 ObjectIDsType) bool {
+	return it.ownerObjectType == it2.ownerObjectType && it.dbTable == it2.dbTable
+}
+
+// ObjectIDsTypes should only be created here
+
+var AddressSetEgressFirewallDNS = newObjectIDsType(addressSet, EgressFirewallDNSOwnerType)
+
+var AddressSetHybridNodeRoute = newObjectIDsType(addressSet, HybridNodeRouteOwnerType)
+
+var AddressSetEgressQoS = newObjectIDsType(addressSet, EgressQoSOwnerType)
+
+var AddressSetNetworkPolicy = newObjectIDsType(addressSet, NetworkPolicyOwnerType)
+
+var AddressSetNamespace = newObjectIDsType(addressSet, NamespaceOwnerType)
+
+var AddressSetEgressIP = newObjectIDsType(addressSet, EgressIPOwnerType)
+
+var AddressSetEgressService = newObjectIDsType(addressSet, EgressServiceOwnerType)
+
+// DbObjectIDs is a structure representing a set of db object ExternalIDs, used to identify
+// an object in the db (as a primary/secondary index) or for a predicate search.
+// DbObjectIDs consists of 3 parts:
+// - IDsType defines which IDs are used for given object, and references a list of ID keys from dbIDsMap map.
+// IDsType.ownerObjectType will be written to ExternalIDs[OwnerTypeKey]
+// - OwnerControllerName defines who manages given object. It is required to communicate with the other
+// controllers that create/update/delete db objects.
+// - ObjectIDs provides values for keys that are used by given IDsType. To create a new object, all fields should be set.
+// For predicate search, only some values that need to be matched may be set.
+//
+//  dbIndex := NewDbObjectIDs(AddressSetEgressFirewallDNS, "DefaultController",
+//		map[string]string{
+//			ObjectNameKey: "dns.name",
+// 			AddressSetIPFamilyKey: "ipv4"
+//	})
+// uses AddressSetEgressFirewallDNS = newObjectIDsType(addressSet, EgressFirewallDNSOwnerType),
+// which in turn uses the following IDs based on dbIDsMap map
+//  dbIDsMap[addressSet][EgressFirewallDNSOwnerType] = {
+//		// dnsName
+//		ObjectNameKey,
+//		AddressSetIPFamilyKey,
+//	}
+//
+// dbIndex will be mapped to the following ExternalIDs
+// {
+//		"k8s.ovn.org/controller": "DefaultController"
+//		"k8s.ovn.org/object-type": "EgressFirewallDNS" (value of EgressFirewallDNSOwnerType)
+//		"k8s.ovn.org/name": "dns.name"
+// 		"k8s.ovn.org/ipFamily": "ipv4"
+//		"k8s.ovn.org/id": "DefaultController:EgressFirewallDNS:dns.name:ipv4"
+// }
+type DbObjectIDs struct {
+	IDsType ObjectIDsType
+	// OwnerControllerName specifies which controller owns the object.
+	// Controller should only change objects it owns, make sure to always set this field.
+	OwnerControllerName string
+	// ObjectIDs are ids that may be different for ObjectIDsType.
+	// These ids should uniquely identify db object with the same OwnerControllerName and OwnerTypeKey.
+	// It may even be empty, in case there is only 1 object of given type in the whole cluster.
+	ObjectIDs map[string]string
```

Comment:
Should we un-export these and provide a getter to make them read-only?

Specifically for ObjectIDs, is it straight forward to understand the difference between ObjectIDs and ExternalIDs or would it benefit to have a GetIDs method with an appropriate godoc text making this difference explicit?

Reply:
ok I did that, now we have some more functions in db_object_ids that were implemented on the caller side before